unified-runtime/source/common/ur_singleton.hpp

```diff
@@ -77,20 +76,22 @@ template <typename singleton_tn, typename key_tn> class singleton_factory_t {
   void retain(key_tn key) {
     std::lock_guard<std::mutex> lk(mut);
     auto iter = map.find(getKey(key));
-    assert(iter != map.end());
-    iter->second.ref_count++;
+    if (iter != map.end()) {
```

Comment:
Won't this (and the later check) cause debug builds to miss invalid keys being used?

Reply:
I'm be more worried about release builds crashing, honestly.